sdk/communication/Azure.Communication.Administration/src/PhoneNumberSearchOperation.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Communication.Administration.Models;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Administration
+{
+    /// <summary>
+    /// Represent a long-running phone number search operation.
+    /// </summary>
+    public class PhoneNumberSearchOperation : Operation<PhoneNumberSearch>
+    {
+        private readonly PhoneNumberAdministrationClient _client;
+        private readonly CancellationToken _cancellationToken;
+        private bool _hasCompleted;
+        private PhoneNumberSearch? _value;
+        private Response _rawResponse;
+        private readonly IReadOnlyList<SearchStatus> _terminateStatuses = new SearchStatus []
+        {
+            SearchStatus.Reserved,
+            SearchStatus.Expired,
+            SearchStatus.Success,
```

Comment:
`Success` means that the number has been purchased, the reserve operation shouldn't wait for that right?